### PR TITLE
Update release docs to include vercel alias

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

Updated the release docs to include the generation of canonical URLs. Since the documentation is growing I moved the relevant bits away from the main `README.md` into the existing `RELEASE.md`.

### :link: External links

Jira ticket: [HDS-3367](https://hashicorp.atlassian.net/browse/HDS-3367)

***
:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

[HDS-3367]: https://hashicorp.atlassian.net/browse/HDS-3367?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/components/RELEASE.md [hunk 1]
```diff
@@ -4,13 +4,40 @@ Whenever there is an update to the components, and these changes have been appro
...
 Please see the instructions in the [CONTRIBUTING](CONTRIBUTING.md) file for more details about how to setup the project and make changes to the code for these packages.
 
-## Bump and release
+## The release process
 
-Follow the instructions for Changesets in the root [README](../../README.md).
+Release PRs, titled 'Version Packages', are created and/or automatically updated on every PR merge by the [changeset GitHub action](https://github.com/changesets/action).
 
-🚨 **DON'T FORGET**:
+In preparation for a release, login to Vercel CLI and create a canonical URL for the upcoming version:
```
Comment (Dhaulagiri) [inline packages/components/RELEASE.md:11]: we should specify that this doesn't apply to patch releases

File: packages/components/RELEASE.md [hunk 2]
```diff
@@ -4,13 +4,40 @@ Whenever there is an update to the components, and these changes have been appro
...
 
-You need to communicate to the product teams that are consuming the components!
+1. Type `vercel login` in your terminal and select 'Continue with SAML Single Sign-On'
```
Comment (didoo) [inline packages/components/RELEASE.md:13]: If I type `vercel login` in my terminal, it says `command not found: vercel`. I imagine we need to install something, can you add a note about what/how should be installed?

**Review by didoo — APPROVED** [review]
Left a comment. Apart from that it's OK for me 👍

File: packages/components/RELEASE.md (hunk 2 above)
Comment (alex-ju) [reply packages/components/RELEASE.md]: I've added an initial step for the Vercel CLI, pointing to their docs (as people may use npm,pnpm, or yarn to manage their global packages.

File: packages/components/RELEASE.md (hunk 1 above)
Comment (alex-ju) [reply packages/components/RELEASE.md]: updated

**Pull request merged**